### PR TITLE
Creating Update User Profile Form

### LINKED ISSUE #108: Update User Profile
- User can update their information in the profile page and update their profile picture.

### PR DESCRIPTION
**Pull request by Leeadar**
- Adding UserUpdateForm and ProfileUpdateForm classes to the forms
- Users can update their information in the profile page and update their profile picture.

> http://127.0.0.1:8000/profile/

image

closes #108

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -22,4 +22,22 @@ def register(request):
 
 @login_required
 def profile(request):
-    return render(request, 'users/profile.html')
+
+    if request.method == 'POST':
+        u_form = UserUpdateForm(request.POST, instance=request.user)
+        p_form = ProfileUpdateForm(
```
Comment (nadavspinco) [inline users/views.py:28]: Please change the variables names from u_form and p_form to user_form and profile_form.

File: users/tests/test_user.py [hunk 2]
```diff
@@ -72,3 +72,11 @@ def test_no_authenticated_client(client, django_user_model):
     response = client.get('/profile/')
     assert response.status_code == 302
     assert response.url == '/login/?next=/profile/'
+
+
+@pytest.mark.django_db
+def test_update_user(user):
```
Comment (Yarboa) [inline users/tests/test_user.py:78]: You already have:
def create_user(db, django_user_model, test_password):
Consider support of calling with fixture test_password or not

File: users/tests/test_user.py [hunk 3]
```diff
@@ -72,3 +72,11 @@ def test_no_authenticated_client(client, django_user_model):
     response = client.get('/profile/')
     assert response.status_code == 302
     assert response.url == '/login/?next=/profile/'
+
+
```
Comment (Yarboa) [inline users/tests/test_user.py:76]: Consider adding form test, 
Consider combining this with the following?
https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Testing#forms

Comment (Leeadar) [reply users/tests/test_user.py:78]: I added the form tests

File: users/tests/test_user.py [hunk 4]
```diff
@@ -72,3 +74,35 @@ def test_no_authenticated_client(client, django_user_model):
...
+        form_data = {
+            'username': 'username',
+            'email': 'Test@example.com',
+            'password1': 'test-password',
+            'password2': 'test-password',
+        }
+        form = UserRegistrationForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+    def test_UserUpdateForm(self):
```
Comment (Shoval-Ovadia) [inline users/tests/test_user.py:100]: I think you should add a test that checks if the update worked (for both - UserUpdate and ProfileUpdate). 
maybe try to save the form and check the fields did update.

**Review by nadavspinco — CHANGES_REQUESTED** [review]
I just notice that the date filed called "dob" on the web page instead of "birth date".

**Comment by Leeadar** [comment]
@nadavspinco I changed the 'dob' field.
WDYT?

image

File: users/views.py (hunk 1 above)
Comment (Leeadar) [reply users/views.py]: Done

File: users/tests/test_user.py (hunk 3 above)
Comment (Yarboa) [reply users/tests/test_user.py:78]: Please refer this PR 
https://github.com/beyond-io/roo.me/blob/main/apartments/tests.py

I would expect using django-pytest

File: users/tests/test_user.py (hunk 4 above)
Comment (Yarboa) [reply users/tests/test_user.py:100]: +2 on that, see comment above related to forms testing

Comment (Leeadar) [reply users/tests/test_user.py]: @Yarboa I added tests for invalid forms and checked that we can't save them.

File: users/tests/test_user.py (hunk 2 above)
Comment (Yarboa) [reply users/tests/test_user.py]: Any response on this?

**Review by Yarboa — COMMENTED** [review]
Please rebase
I have got this:
image

**Comment by Yarboa** [comment]
> Please rebase
> I have got this:
> image

@nadavspinco 
This is the reason why Post/tests/test_url_posts.py: is not valuable
There is broken link, and tests did not receive it

**Comment by Leeadar** [comment]
@Yarboa I rebased, is it works now ?

**Comment by Yarboa** [comment]
> @Yarboa I rebased, is it works now ?

Yes

File: users/tests/test_user.py (hunk 2 above)
Comment (Yarboa) [reply users/tests/test_user.py]: > Any response on this?

OK i see it was  changed

**Review by Yarboa — APPROVED** [review]
LGTMT +2

**Pull request merged**